### PR TITLE
Role Distribution

### PR DESCRIPTION
**Pull request by ZakMiller**

### CODE REVIEW

File: public/scripts/roles.js [hunk 1]
```diff
@@ -1,25 +1,19 @@
-class Role {
-  constructor(image, name, description) {
-    this.image = image
-    this.name = name
-    this.description = description
+const roles = {
+  human: {
+    image: 'https://s-media-cache-ak0.pinimg.com/originals/e0/f8/41/e0f84168cde4123e1e5aec348364a017.jpg',
+    name: 'Human',
+    description: "You can say whatever you want, but be careful! If the robots figure out you're human it 's all over!"
```
Comment (SylvanSign) [inline public/scripts/roles.js]: It 's

File: server.js [hunk 2]
```diff
@@ -44,18 +72,8 @@ function handleSocketConnection(socket) {
...
-          io.to(currentRoom).emit('stateChange', 'game')
-          console.log(`${currentRoom} started a game!`)
-          currentRoom = generateRoomName()
-          users.clear()
-          io.emit('lobbyUpdate', users.list())
-        }
-      }, START_GAME_DELAY * ONE_SECOND + DELAY_BUFFER_IN_MS)
-      io.to(currentRoom).emit('countdown', START_GAME_DELAY)
+      startGame()
+      assignRoles(currentRoom)
```
Comment (SylvanSign) [inline server.js]: This is in the wrong place. We only want to assign roles after the game state event has been emitted, so it should be executed some time right after a successful countdown and a start of game

File: server.js [hunk 3]
```diff
@@ -25,6 +25,34 @@ app.use(express.static(path.join(__dirname, 'public')))
...
+      currentRoom = generateRoomName()
+      users.clear()
+      io.emit('lobbyUpdate', users.list())
+    }
+  }, START_GAME_DELAY * ONE_SECOND + DELAY_BUFFER_IN_MS)
+  io.to(currentRoom).emit('countdown', START_GAME_DELAY)
+}
+
+function assignRoles(currentRoom) {
+  io.in(currentRoom).clients((err, clients) => {
```
Comment (SylvanSign) [inline server.js:46]: `.in` and `.to` mean the same, but I think I like `.to` better, and you use it more often as well. Can we stick to just that one?

File: public/scripts/main.js [hunk 4]
```diff
@@ -162,3 +150,20 @@ const gameHandler = (function gameIIFE() {
 })()
 
 stateManager.register('game', gameHandler)
+
+socket.on('setRole', setRole)
+
+function displayRole(role) {
```
Comment (SylvanSign) [inline public/scripts/main.js:154]: You can destructive this parameter into the three fields you need below. That's optional tho

**Review by SylvanSign — CHANGES_REQUESTED** [review]

File: public/scripts/main.js (hunk 4 above)
Comment (SylvanSign) [reply public/scripts/main.js:154]: Destructure

File: public/scripts/main.js [hunk 5]
```diff
@@ -145,9 +136,6 @@ const gameHandler = (function gameIIFE() {
     game.hidden = false
     const STARTING_REQUIRED_WORD_COUNT = 10
     requiredWords.addWords(STARTING_REQUIRED_WORD_COUNT)
```
Comment (SylvanSign) [inline public/scripts/main.js]: This adding of words is only needed for the AI, right? Can you fix this up as part of these changes?

File: server.js (hunk 3 above)
Comment (ZakMiller) [reply server.js:46]: I think it depends on the situation...
Here I think 'in' makes sense. You are saying io. in current room . clients. In other situations I think 'to' makes sense.

File: public/scripts/main.js (hunk 5 above)
Comment (ZakMiller) [reply public/scripts/main.js]: Ya good point. Hadn't thought of that.

File: server.js (hunk 2 above)
Comment (ZakMiller) [reply server.js]: How would you do it?

Comment (SylvanSign) [reply server.js]: Put it within `startGame()`, but only call it after the game has actually started, not when we've started the countdown to a game. You're just calling it in the wrong place.

File: server.js (hunk 3 above)
Comment (SylvanSign) [reply server.js:46]: I gotcha. Fine with me then

File: public/scripts/main.js (hunk 5 above)
Comment (ZakMiller) [reply public/scripts/main.js]: Done

**Review by SylvanSign — APPROVED** [review]

**Comment by SylvanSign** [comment]
Ready for merge?

**Pull request merged**